lib/core/prometheusMetrics.cpp

```diff
@@ -55,13 +55,18 @@ void Gauge::set(const double value) {
 }
 
 string Gauge::to_string() {
+    if (std::isnan(value)) {
+        return fmt::format(fmt("{} {:d}"), "NaN", last_update_time_stamp);
+    } else if (std::isinf(value)) {
+        return fmt::format(fmt("{} {:d}"), (value < 0 ? "-Inf" : "+Inf"), last_update_time_stamp);
+    }
     return fmt::format(fmt("{:f} {:d}"), value, last_update_time_stamp);
 }
 
 std::ostringstream& Gauge::to_string(std::ostringstream& out) {
     std::lock_guard<std::mutex> lock(metric_lock);
 
-    fmt::print(out, fmt("{:f} {:d}"), value, last_update_time_stamp);
+    fmt::print(out, this->to_string());
```

Comment:
I think the reason this line was duplicated in both methods is because the `to_string()` one might allocate memory for the string and then it's freed when it's just copied to `ostringstream` (which presumably already has memory allocated for it, so you're just copying bytes at the end). If anything, do it the other way around: implement the full "if" in `to_string(ostringstream&) and then in `to_string()` just do:
```
 std::ostringstream buf;
to_string(buf);
return buf.str();
```

Reply:
This is just a small optimization, it might not make a huge difference. But the "to_string()" might be just a convenience I added for debugging. In the `serialize()` method, which is what gets called on a scrape to `/metrics`, it just iterates through all known metrics and calls the `to_string(ostringstream&)` method, joining them all with newlines.

Of course, now I see that in prometheusMetrics.cpp, I do the "ostring.str()" in a number of places, which kind of defeats the purpose. I should go back and clean that up.

Reply:
Okay!